src/imap.rs

```diff
@@ -1641,39 +1646,64 @@ impl Imap {
     }
 }
 
-/// Returns target folder for a message found in the Spam folder.
-async fn spam_target_folder(
+async fn should_move_out_of_spam(
     context: &Context,
-    folder: &str,
     headers: &[mailparse::MailHeader<'_>],
-) -> Result<Option<Config>> {
-    if let Some(chat) = prefetch_get_chat(context, headers).await? {
-        if chat.blocked != Blocked::Not {
+) -> Result<bool> {
+    if headers.get_header_value(HeaderDef::ChatVersion).is_some() {
+        // If this is a chat message (i.e. has a ChatVersion header), then this might be
+        // a securejoin message. We can't find out at this point as we didn't prefetch
+        // the SecureJoin header. So, we always move chat messages out of Spam.
+        // One possibility to change this woudl be:
+        // Remove the `&& !context.is_spam_folder(folder).await?` check from
+        // `fetch_new_messages()`, and then let `dc_receive_imf()` check
+        // if it's a spam message and should be hidden.
+        return Ok(true);
```

Comment:
my only reservation here is that spammers now simply have to add a ChatVersion header and whatever the spam filter does no longer matters as deltachat will move it back.  This is not currently a problem, but one day it probably will be.

I think at that time we can start to do more smarts here anyway, so i guess it's fine for now.

Reply:
Yup, this was also my reasoning.